tutorial/SparkRedshiftTutorial.scala

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2015 Databricks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.databricks.spark.redshift.tutorial
+
+import org.apache.spark.{ SparkConf, SparkContext }
+import org.apache.spark.sql.SaveMode
+import java.io.File
+import org.apache.spark.sql.types.{ StructType, StructField, 
+                                    StringType, IntegerType, 
+                                    LongType, DecimalType }
+
+/**
+ * Source code accompanying the spark-redshift tutorial. 
+ * The following parameters need to be passed
+ * 1. AWS Access Key
+ * 2. AWS Secret Access Key
+ * 3. Redshift Database Name
+ * 4. Redshift UserId
+ * 5. Redshift Password
+ * 6. Redshift URL (Ex. swredshift.czac2vcs84ci.us-east-1.redshift.amazonaws.com:5439
+ * 
+ */
+object SparkRedshiftTutorial {
+  /*
+   * For Windows Users only
+   * 1. Download contents from link 
+   *      https://github.com/srccodes/hadoop-common-2.2.0-bin/archive/master.zip
+   * 2. Unzip the file in step 1 into your %HADOOP_HOME%/bin. 
+   *        In the example below  %HADOOP_HOME%=C:/MySoftware/hdphome
+   * 3. After step 2 %HADOOP_HOME/bin should contain winutils.exe
+   * 
+   */   
+	if (System.getProperty("os.name").startsWith("Windows")) {
+		val hdphome: File = new File("C:/MySoftware/hdphome");
+		System.setProperty("hadoop.home.dir", hdphome.getAbsolutePath().toString())
+	}
+
+	def main(args: Array[String]): Unit = {
+		val awsAccessKey = args(0)
+		val awsSecretKey = args(1)
+		val rsDbName = args(2)
+		val rsUser = args(3)
+		val rsPassword = args(4)
+    /** Sample Redshift URL is swredshift.czac2vcs84ci.us-east-1.redshift.amazonaws.com:5439 */
+		val rsURL = args(5)  
+		val jdbcURL = s"""jdbc:redshift://$rsURL/$rsDbName?user=$rsUser&password=$rsPassword"""
+    println(jdbcURL)
+		val sc = new SparkContext(new SparkConf().setAppName("SparkSQL").setMaster("local"))
+
+		val tempS3Dir = "s3n://redshift-spark/temp/"
```

Comment:
I removed this while improving on the comment. My bad. It is enough to pass -D-Dhadoop.home.dir=C:/MySoftware/hdphome in my case. I removed it. Looks cleaner now